### PR TITLE
chore: add utils CRAP score check

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context

follow up https://github.com/PostHog/posthog-python/pull/576

Follow-up to the targeted mutation-testing PR: add a CRAP-score gate for `posthog/utils.py` using `pytest-crap` instead of a hardcoded CRAP formula. This keeps the complexity/coverage quality bar explicit in CI while avoiding custom reimplementation of CRAP scoring.

## :green_heart: How did you test it?

- `uv run --extra test --with pytest-crap --with pytest-cov pytest posthog/test/test_utils.py posthog/test/test_size_limited_dict.py --timeout=30 --cov=posthog.utils --crap --crap-threshold=10 -q`
- `uv run --extra test --with pytest-crap --with pytest-cov python .github/scripts/check_crap_threshold.py posthog/utils.py --max-crap 10`
- `rm -f .coverage && rm -rf mutants && uv run --extra test --with mutmut mutmut run --max-children 1 && uv run --extra test --with mutmut mutmut results`
- `uv run --extra test pytest --timeout=30 -q`
- `uvx ruff check .github/scripts/check_crap_threshold.py posthog/utils.py posthog/test/test_utils.py posthog/test/test_size_limited_dict.py`
- `ruby -e 'require "yaml"; YAML.load_file(".github/workflows/ci.yml"); puts "ci yaml ok"'`

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [x] No breaking change or entry added to the changelog.

### If releasing new changes

- [ ] Ran `sampo add` to generate a changeset file

### CODE REVIEW

File: posthog/test/test_utils.py [hunk 1]
```diff
@@ -681,6 +713,16 @@ def test_redis_errors_fall_back_to_miss(self):
         failing_cache.invalidate_version(1)
         failing_cache.clear()
 
+    def test_redis_cache_key_helpers(self):
+        key = self.cache._get_cache_key("user123", "missing")
+        invalid_key = self.cache._get_cache_key("user123", "invalid")
+        self.redis.store[invalid_key] = "not json"
+
+        assert self.cache._redis_key_to_string(key) == key
+        assert self.cache._key_has_version(key, 1) is False
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply posthog/test/test_utils.py]: Fixed in 40b2ded — split the Redis key/version helper coverage into focused tests and added an explicit corrupt-entry deletion test.

File: posthog/test/test_utils.py [hunk 2]
```diff
@@ -327,6 +346,10 @@ def test_get_os_info_branches(self):
         ):
             assert utils.get_os_info() == {"$os": "sunos", "$os_version": "5.11"}
 
+        with mock.patch.object(utils.platform, "release", create=True):
+            delattr(utils.platform, "release")
+            assert utils._platform_release() == ""
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply posthog/test/test_utils.py]: Fixed in 40b2ded — split the OS helper edge cases into focused parameterized tests.

File: posthog/utils.py [hunk 3]
```diff
@@ -409,6 +397,32 @@ def invalidate_version(self, old_version):
...
+    def _key_has_version(self, key, old_version):
+        try:
+            data = self.redis.get(key)
+            if not data:
+                return False
+            return json.loads(data).get("flag_version") == old_version
+        except (json.JSONDecodeError, KeyError):
+            # If we can't parse the entry, delete it to be safe
+            self.redis.delete(key)
+            return False
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply posthog/utils.py:432]: Fixed in 40b2ded — moved corrupt-cache deletion out of `_key_has_version` and into `_delete_keys_with_version` so the predicate no longer deletes as a side effect.

**Comment by marandaneto** [comment]
merged https://github.com/PostHog/posthog-python/pull/578 into this branch so you can skip the acceptance tests review, forgot about the stacked PRs, sorry

**Review by dustinbyrne — APPROVED** [review]
